gravitee-plugin-core/src/main/java/io/gravitee/plugin/core/api/AbstractConfigurablePluginManager.java

```diff
@@ -50,15 +51,20 @@ public String getSchema(String pluginId, String subFolder) throws IOException {
     public String getIcon(String pluginId) throws IOException {
         T plugin = get(pluginId);
         Map<String, String> properties = plugin.manifest().properties();
-        if (properties != null) {
-            String icon = properties.get(PluginManifestProperties.MANIFEST_ICON_PROPERTY);
-            if (icon != null) {
-                Path iconFile = Paths.get(plugin.path().toString(), icon);
-                String mimeType = Files.probeContentType(iconFile);
-                return "data:" + mimeType + ";base64," + Base64.getEncoder().encodeToString(Files.readAllBytes(iconFile));
+        return this.getFileFromPropertyAsBase64(plugin, properties, PluginManifestProperties.MANIFEST_ICON_PROPERTY);
+    }
+
+    private String getFileFromPropertyAsBase64(T plugin, Map<String, String> properties, String property) throws IOException {
+        if (properties != null && properties.containsKey(property)) {
+            String path = properties.get(property);
+            try {
+                Path file = Paths.get(plugin.path().toString(), path);
+                String mimeType = Files.probeContentType(file);
+                return "data:" + mimeType + ";base64," + Base64.getEncoder().encodeToString(Files.readAllBytes(file));
+            } catch (NoSuchFileException ex) {
+                return path;
```

Comment:
So we keep this behavior ? no error and return the value of the props

Reply:
No, I will remove, it's useless now